runChecks.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+
+"""
+Replacement for test-math-dependencies target in Makefile.
+
+"""
+
+from __future__ import print_function
+import os
+import os.path
+import sys
+import re
+import glob
+
+winsfx = ".exe"
+testsfx = "_test.cpp"
+def files_in_folder(folder):
+    files = []
+    for f in glob.glob(folder):
+        if os.path.isdir(f):
+            files += files_in_folder(f+os.sep+"**")
+        else:
+            files.append(f)
+    return files
+def grep_patterns(type, folder, patterns_and_messages, exclude_filters = []):
+    """Checks the files in the provided folder for matches
+    with any of the patterns. It returns an array of
+    messages and the provided type with
+    the line number. This check ignores comments.
+    """
+    errors = []
+    folder.replace("/", os.sep)
+    exclude_files = []
+    for excl in exclude_filters:
+        exclude_files += files_in_folder(excl)
+    files = files_in_folder(folder+os.sep+"**")
+    files = [x for x in files if x not in exclude_files]
+    for f in files:
+        if os.path.isfile(f):
+            line_num = 0
+            for line in open(f, "r"):
+                line_num += 1
+                
+                for p in patterns_and_messages:
+                    # exclude line starting with "/*" or " * " and
+                    # if the matched patterns are behind "//"
+                    if not re.search("^ \* |^/\*", line) and not re.search(".*//.*"+p["pattern"], line) and re.search(p["pattern"], line):
+                            errors.append(f + " at line " + str(line_num) + ":\n\t" + "[" + type + "] " + p["message"])
+    return errors
+
+def check_non_test_files_in_test():
+    all_cpp = files_in_folder("test/unit/*/*")
```

Comment:
`all_cpp = files_in_folder("test/unit")`

Reply:
This one was done because I want to ignore the `test/unit` files where we have the `translation_unit` cpp files that are the only non `_test.cpp` files we have. It was either that or a list of exception files, this seems cleaner.

Reply:
I changed it to `"test/unit/math"`. Thanks